yb-voyager/src/anon/sql_anonymizer.go

```diff
@@ -0,0 +1,510 @@
+			ie.Name, err = a.registry.GetHash(COLUMN_KIND_PREFIX, ie.Name)
+			if err != nil {
+				return fmt.Errorf("anon index column name: %w", err)
+			}
+		}
+
+		// 2) Expression index: (Expr != nil, Name == "")
+		//    CREATE INDEX … ON tbl((col1+col2));
+		if ie.Expr != nil {
+			// Walk into the expression node and anonymize any ColumnRefs inside.
+			// Assuming you already handle ColumnRef elsewhere in your visitor:
+			//   queryparser.TraverseParseTree(ie.Expr, visited, a.anonymizationProcessor)
+		}
+
+		// 3) Expression alias: (Indexcolname != "")
+		//    CREATE INDEX … ON tbl((col1+col2) AS sum_col);
+		//	 above this sql syntax is invalid, couldn't find an example of Indexcolname but still keeping the anonymization logic here
+		if ie.Indexcolname != "" {
+			ie.Indexcolname, err = a.registry.GetHash(ALIAS_KIND_PREFIX, ie.Indexcolname)
+			if err != nil {
+				return fmt.Errorf("anon index column alias: %w", err)
+			}
+		}
+
+	/*
+		SQL:		ALTER TABLE ONLY public.foo ADD CONSTRAINT unique_1 UNIQUE (column1, column2) DEFERRABLE;
+		ParseTree:	stmt:{alter_table_stmt:{relation:{schemaname:"public" relname:"foo" relpersistence:"p" } cmds:{alter_table_cmd:{subtype:AT_AddConstraint
+					def:{constraint:{contype:CONSTR_UNIQUE conname:"unique_1" deferrable:true location:32 keys:{string:{sval:"column1"}}
+					keys:{string:{sval:"column2"}}}} behavior:DROP_RESTRICT}} ...}}
+	*/
+	case queryparser.PG_QUERY_CONSTRAINT_NODE:
+		cons, err := queryparser.ProtoAsTableConstraintNode(msg)
+		if err != nil {
+			return err
+		}
+		if cons.Conname != "" {
+			cons.Conname, err = a.registry.GetHash(CONSTRAINT_KIND_PREFIX, cons.Conname)
+			if err != nil {
+				return fmt.Errorf("anon constraint: %w", err)
+			}
+		}
+
+		if len(cons.Keys) > 0 {
+			// For each key in the constraint, anonymize the column names
+			for i, key := range cons.Keys {
+				if key.GetString_() == nil {
+					continue
+				}
+				colName := key.GetString_().Sval
+				if colName == "" {
+					continue // skip empty names
+				}
+				key.GetString_().Sval, err = a.registry.GetHash(COLUMN_KIND_PREFIX, colName)
+				if err != nil {
+					return fmt.Errorf("anon constraint key[%d]=%q: %w", i, colName, err)
+				}
+			}
+		}
+
+	/*
+		ALTER TABLE humanresources.department CLUSTER ON \"PK_Department_DepartmentID\";
+		stmt: {alter_table_stmt:{relation:{schemaname:"humanresources" relname:"department" ...}
+			cmds:{alter_table_cmd:{subtype:AT_ClusterOn name:"PK_Department_DepartmentID" behavior:...}} objtype:OBJECT_TABLE}}
+	*/
+	case queryparser.PG_QUERY_ALTER_TABLE_STMT:
+		ats, ok := queryparser.ProtoAsAlterTableStmtNode(msg)
+		if !ok {
+			return fmt.Errorf("expected AlterTableStmt, got %T", msg.Interface())
+		}
+
+		for _, cmd := range ats.Cmds {
+			alterTableCmdNode := cmd.GetAlterTableCmd()
+			if alterTableCmdNode == nil {
+				continue // skip if not an AlterTableCmd
+			}
+
+			if alterTableCmdNode.GetSubtype() == pg_query.AlterTableType_AT_ClusterOn {
```

Comment:
what about other atler tables in general? they table names, column names get covered with the other switch cases?